core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -280,11 +282,23 @@ private void addTablesRecords(QualifiedTablePrefix prefix)
     {
         Set<SchemaTableName> tables = listTables(session, metadata, accessControl, prefix);
         Set<SchemaTableName> views = listViews(session, metadata, accessControl, prefix);
+        Set<SchemaTableName> mviews = listMaterializedViews(session, metadata, accessControl, prefix);
+        Set<SchemaTableName> storageTables = listStorageTab(session, metadata, accessControl, prefix);
         // TODO (https://github.com/trinodb/trino/issues/8207) define a type for materialized views
 
         for (SchemaTableName name : union(tables, views)) {
             // if table and view names overlap, the view wins
-            String type = views.contains(name) ? "VIEW" : "BASE TABLE";
+            String type = "BASE TABLE";
+            if (views.contains(name)) {
+                type = "VIEW";
+            }
+            else if (mviews.contains(name)) {
+                type = "MATERIALIZE VIEW";
```

Comment:
This was left as a TODO (see a few lines above https://github.com/trinodb/trino/pull/10745/files#diff-de0b2e0ffdaf5c2eeda3ad7f0513ad28373a158a1f4fbe32ef0a10f5ce10cc73L283) because information schema table is supposed to follow ANSI SQL spec and we didn't have clarity about which table type should be used for MVs if it is to be differentiated from the others.
fyi @sopel39 @findepi @martint

Reply:
If we want to follow spec, we should return "VIEW" for materialized views.
but let's wait maybe @martint has more thoughts on this.

Reply:
materialize view and normal view is different, so i think they are different type